src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -73,18 +98,50 @@ public CommandResult execute(Model model) throws CommandException {
         if (index.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
         }
-
         Person personToEdit = lastShownList.get(index.getZeroBased());
-        Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor);
 
-        if (!personToEdit.isSamePerson(editedPerson) && model.hasPerson(editedPerson)) {
-            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
-        }
+        if (personToEdit instanceof Student) {
+
+            Student editedStudent;
+            Student studentToEdit = (Student) personToEdit;
+            if (editStudentDescriptor != null) {
+                editedStudent = createEditedStudent(studentToEdit, editStudentDescriptor);
+            } else {
+                EditStudentDescriptor editStudentDescriptor = new EditStudentDescriptor(editPersonDescriptor);
+                editedStudent = createEditedStudent(studentToEdit, editStudentDescriptor);
+            }
+
+            // if you are changing the name to one that already exists
+            if (!studentToEdit.isSamePerson(editedStudent) && model.hasPerson(editedStudent)) {
+                throw new CommandException(MESSAGE_DUPLICATE_PERSON);
+            }
+
+            if (!model.hasTutorialWithName(editedStudent.getTutorialName())) {
+                throw new CommandException(String.format(MESSAGE_TUTORIAL_DOES_NOT_EXIST,
```

Comment:
maybe you can use the MESSAGE_TUTORIAL_NOT_FOUND in common.core.Messages instead? for standardisation :))

Reply:
Ok i think for standardisation i can use this, only thing is that the message in Messages does not tell the user the specific tutorial name that does not exist.

Reply:
so for the other classes i did, like removeStudentCommand, I will continue to use the current message because it has more detail and may help us with debugging later on.